test/disjointCoarseMallocPool.cpp

```diff
@@ -0,0 +1,610 @@
+    umfMemoryProviderDestroy(malloc_memory_provider);
+}
+
+TEST_F(test, disjointCoarseMallocPool_simple2) {
+
+    umf_memory_provider_handle_t malloc_memory_provider;
+    umfMemoryProviderCreate(&UMF_MALLOC_MEMORY_PROVIDER_OPS, NULL,
+                            &malloc_memory_provider);
+    ASSERT_NE(malloc_memory_provider, nullptr);
+
+    const size_t KB = 1024;
+    const size_t MB = 1024 * KB;
+
+    const size_t init_buffer_size = 20 * MB;
+
+    coarse_memory_provider_params_t coarse_memory_provider_params = {
+        malloc_memory_provider, // upstream_memory_provider
+        true,                   // immediate_init
+        0,
+        init_buffer_size,
+        true,  // trace
+        false, // WA_do_not_free_upstream
+    };
+
+    umf_memory_provider_handle_t coarse_memory_provider;
+    umfMemoryProviderCreate(umfCoarseMemoryProviderOps(),
+                            &coarse_memory_provider_params,
+                            &coarse_memory_provider);
+    ASSERT_NE(coarse_memory_provider, nullptr);
+
+    umf_disjoint_pool_params_t disjoint_memory_pool_params = {};
+    disjoint_memory_pool_params.SlabMinSize = 4096;
+    disjoint_memory_pool_params.MaxPoolableSize = 4096;
+    disjoint_memory_pool_params.Capacity = 4;
+    disjoint_memory_pool_params.MinBucketSize = 64;
+    disjoint_memory_pool_params.PoolTrace = 1;
+
+    umf_memory_pool_handle_t pool;
+    umfPoolCreate(umfDisjointPoolOps(), coarse_memory_provider,
+                  &disjoint_memory_pool_params, UMF_POOL_CREATE_FLAG_NONE,
+                  &pool);
+    ASSERT_NE(pool, nullptr);
+
+    // test
+    double sizes[] = {2, 4, 0.5, 1, 8, 0.25};
+    size_t alignment[] = {0, 4, 0, 16, 32, 128};
+    for (int i = 0; i < 6; i++) {
+        size_t s = (size_t)(sizes[i] * MB);
+        void *t[8] = {0};
+        for (int j = 0; j < 8; j++) {
+            t[j] = umfPoolAlignedMalloc(pool, s, alignment[i]);
+            ASSERT_NE(t[j], nullptr);
+        }
+
+        for (int j = 0; j < 8; j++) {
+            umf_result_t res = umfPoolFree(pool, t[j]);
+            ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+        }
+    }
+
+    umfPoolDestroy(pool);
+    umfMemoryProviderDestroy(coarse_memory_provider);
+    umfMemoryProviderDestroy(malloc_memory_provider);
+}
+
+struct alloc_ptr_size {
+    void *ptr;
+    size_t size;
+
+    bool operator<(const alloc_ptr_size &other) const {
+        if (ptr == other.ptr) {
+            return size < other.size;
+        }
+        return ptr < other.ptr;
+    }
+};
+
+TEST_F(test, disjointCoarseMMapPool_random) {
+
+    const size_t KB = 1024;
+    const size_t MB = 1024 * KB;
+    const size_t init_buffer_size = 200 * MB;
+
+    // Preallocate some memory
+    void *buf = malloc(init_buffer_size);
+    memset(buf, 0, init_buffer_size);
+    ASSERT_NE(buf, nullptr);
+
+    const unsigned char alloc_check_val = 11;
+    coarse_memory_provider_params_t coarse_memory_provider_params = {
+        NULL,  // upstream_memory_provider
```

Comment:
Can't you just set each field separately as you do with disjoint_pool_params below instead of using comments?